Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp

```diff
@@ -1098,6 +1098,19 @@ void NetworkResourceLoader::didFinishLoading(const NetworkLoadMetrics& networkLo
     cleanup(LoadResult::Success);
 }
 
+void NetworkResourceLoader::sendDidFinishResourceLoad(const NetworkLoadMetrics& networkLoadMetrics)
+{
+    if (m_didChangeOfNetworkLoad && m_parameters.options.mode != FetchOptions::Mode::Navigate && !networkLoadMetrics.failsTAOCheck) {
+        auto updatedNetworkLoadMetrics = networkLoadMetrics;
+        updatedNetworkLoadMetrics.redirectCount = m_redirectCount;
```

Comment:
This PR seems to update redirectCount but not redirectStart like the title suggests.  Is that also needed?